tests/test_s3_to_glue_lambda.py

```diff
@@ -1,70 +1,291 @@
                 "object": {
-                  "key": "main/2023-01-12T22--02--17Z_77fefff8-b0e2-4c1b-b0c5-405554c92368",
-                  "size": 1024,
-                  "eTag": "0123456789abcdef0123456789abcdef",
-                  "sequencer": "0A1B2C3D4E5F678901"
-                }
-              }
-            }
-          ]
+                    "key": "main/2023-01-12T22--02--17Z_77fefff8-b0e2-4c1b-b0c5-405554c92368",
+                    "size": 1024,
+                    "eTag": "0123456789abcdef0123456789abcdef",
+                    "sequencer": "0A1B2C3D4E5F678901",
+                },
+            },
         }
         return s3_event
 
     @pytest.fixture
-    def test_messages(self):
-        messages = [
+    def test_sqs_message(self, test_s3_event):
+        sqs_msg = {
+            "Records": [
                 {
-                    "source_bucket": "recover-dev-input-data",
-                    "source_object": "main/2023-01-12T22--02--17Z_77fefff8-b0e2-4c1b-b0c5-405554c92368"
+                    "MessageId": "string",
+                    "receiptHandle": "string",
+                    "MD5OfBody": "string",
+                    "body": "string",
+                    "Attributes": {
+                        "string": "string",
+                    },
+                    "MD5OfMessageAttributes": "string",
+                    "MessageAttributes": {
+                        "string": {
+                            "DataType": "string",
+                            "StringValue": "string",
+                            "BinaryValue": "string",
+                        }
+                    },
                 }
+            ]
+        }
+        sqs_msg["Records"][0]["body"] = json.dumps({"Records": [test_s3_event]})
+        yield sqs_msg
+
+    @pytest.fixture
+    def test_object_info(self):
+        object_info = [
+            {
+                "source_bucket": "recover-dev-input-data",
+                "source_key": "main/2023-01-12T22--02--17Z_77fefff8-b0e2-4c1b-b0c5-405554c92368",
+            }
         ]
-        return messages
+        return object_info
+
+    @pytest.fixture
+    def test_set_env_var(self, monkeypatch, test_sqs_queue_url):
+        monkeypatch.setenv("SQS_QUEUE_URL", test_sqs_queue_url)
+        monkeypatch.setenv("PRIMARY_WORKFLOW_NAME", "test_workflow")
 
-    def test_submit_s3_to_json_workflow(self, test_messages, monkeypatch):
+    def test_submit_s3_to_json_workflow(self, test_object_info, monkeypatch):
         monkeypatch.setattr("boto3.client", lambda x: MockGlueClient())
         app.submit_s3_to_json_workflow(
-                messages=test_messages,
-                workflow_name="example-workflow"
+            objects_info=test_object_info, workflow_name="example-workflow"
         )
+
+    @mock_sqs
+    def test_that_lambda_handler_does_not_call_submit_s3_to_json_workflow_if_empty_sqs_event(
+        self, empty_queue, test_sqs_queue_url, test_set_env_var
+    ):
```

Comment:
This isn't something we have to test because it's not part of our application logic. This is something AWS should be worried about.

Reply:
I think this is just to verify the if no `Records' key in our sqs event message logic in our code unless you are suggesting that we remove that code as well?

Reply:
Yes, we can assume that the message is well-formatted.

Reply:
Hmmm I just realized this. The reason I checked for `Records` key in `s3_event_records` and added this test because I ran into an `KeyError` before when testing the lambda after dropping s3 files indicating there were some things in the SQS message that seemed to be related to S3 events but not contain records..

Reply:
For some reason there's a S3 test event that gets thrown into the bag: 
```
{'Service': 'Amazon S3', 'Event': 's3:TestEvent', 'Time': '2023-09-21T19:43:08.535Z', 'Bucket': 'recover-dev-input-data', 'RequestId': '9YF3QTRANHQ28EY0', 'HostId': 'CXkCznSHquU8oTyCXMJW/gQsatZVAjbvs0YzKxtWYxJD6+QbpZym0az8nW8xJvT3zzRHUt3IuQ0='}
```

Reply:
https://www.mikulskibartosz.name/what-is-s3-test-event